src/common/src/types/mod.rs

```diff
@@ -86,8 +86,13 @@ pub enum DataType {
     Timestamp,
     Timestampz,
     Interval,
-    Struct { fields: Arc<[DataType]> },
-    List { datatype: Box<DataType> },
+    Struct {
+        fields: Arc<[DataType]>,
+        field_names: Arc<[String]>,
```

Comment:
Can we merge them into a single field? This will reduce 16 bytes for every `DataType` instance.
```suggestion
        fields: Arc<[(DataType, String)]>,
```

Reply:
Agree, I was considering making `field_names` optional since in the execution layer we will no longer use it. But so far it isn't left empty in any case because it will increase the difficulty to test. So let's unify them into one field.

Reply:
Oh, I forgot about the case, like `ROW(1, 2)`, where the field names are unnamed. A DataType may not always have a name.

Reply:
> Oh, I forgot about the case, like `ROW(1, 2)`, where the field names are unnamed. A DataType may not always have a name.

 `fields: Arc<[(DataType, Option<String>)]>`? maybe we need support ROW(a, 1, b, 2)

Reply:
or just give it some default name when it is not named.

Reply:
Fixed now. It is replaced with:

```
pub struct StructType {
    pub fields: Vec<DataType>,
    pub field_names: Vec<String>,
}

#[derive(Debug, Clone, PartialEq, Eq, Hash)]
pub enum DataType {
    ...
    Struct(Arc<StructType>),
}

```

Reply:
Why not use tuple here? Is there any case that the length of `fields` are not equal to `field_names`? 👀

Reply:
Like I mentioned above: `ROW(1,2)`. Postgres allows such an "anonymous tuple". In this case, `field_names` will be empty. (I don't like the idea to give each field an empty String name, which takes extra memories as well.

Reply:
If the lengths are not equal but also not empty, this will be confusing. May make the whole field optional.

Reply:
another question comes into my mind: is the DataType on the critical path? now the answer is yes because we use Datum everywhere(even the Row is Vec<Datum>). But it should not be like it and in an ideal relational execution, all DataType is just appear in the schema on the Executor, Chunk, or other data structures.

Reply:
Yes. It's not **THAT** critical but it will still be necessary to ensure its memory efficiency.